externals/kyuubi-spark-sql-engine/src/main/scala/org/apache/kyuubi/engine/spark/operation/ExecuteStatement.scala

```diff
@@ -112,17 +112,19 @@ class ExecuteStatement(
         }
       }
 
+      val sparkSQLSessionManager = session.sessionManager
       try {
-        val sparkSQLSessionManager = session.sessionManager
         val backgroundHandle = sparkSQLSessionManager.submitBackgroundOperation(asyncOperation)
         setBackgroundHandle(backgroundHandle)
       } catch {
-        case rejected: RejectedExecutionException =>
+        case rejected: RejectedExecutionException if !sparkSQLSessionManager.isShutdown =>
           setState(OperationState.ERROR)
           val ke = KyuubiSQLException("Error submitting query in background, query rejected",
```

Comment:
how about we refine the error message to show that the engine is in shutting down phase.